### PR TITLE
literature: add search result items

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds components for search result items.

### CODE REVIEW

File: src/common/components/Latex/__tests__/Latex.test.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    const wrapper = shallow((
+      <Latex>
+        This does not have latex in it
+      </Latex>
+    ));
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('renders text with LaTex in $...$', () => {
+    const formula = '2.8~\\mathrm{fb}^{-1}';
```
Comment (drjova) [inline src/common/components/Latex/__tests__/Latex.test.jsx]: I'd add a more real example from our existing records

File: src/literature/citeArticle.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  try {
+    const response = await http.get(`/literature/${recordId}`, {
+      headers: {
+        Accept: `application/x-${format}`,
+      },
+    });
+    return response.data;
+  } catch (error) {
+    // TODO: handle error better maybe?
+    return `nothing for format: ${format}`;
```
Comment (drjova) [inline src/literature/citeArticle.js]: we can have a reducer for notifications

Comment (drjova) [reply src/literature/citeArticle.js]: but we can add it later

File: src/literature/components/__tests__/LiteratureItem.test.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { fromJS } from 'immutable';
+
+import LiteratureItem from '../LiteratureItem';
+
+describe('LiteratureItem', () => {
+  it('renders with all props set, including sub props', () => {
+    const metadata = fromJS({
```
Comment (drjova) [inline src/literature/components/__tests__/LiteratureItem.test.jsx]: why not adding a record fixture so we can easily reuse it?

File: src/common/components/Latex/__tests__/Latex.test.jsx (hunk 1 above)
Comment (harunurhan) [reply src/common/components/Latex/__tests__/Latex.test.jsx]: Actually records can have anything including this, there is no pattern here.

File: src/literature/citeArticle.js (hunk 2 above)
Comment (harunurhan) [reply src/literature/citeArticle.js]: yes, I agree. When we start giving for more important stuff, we could just plug this in too.

File: src/literature/components/LiteratureItem.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import ReportNumberList from './ReportNumberList';
+import PublicationInfoList from './PublicationInfoList';
+import ArxivPdfDownloadAction from './ArxivPdfDownloadAction';
+import CiteModalAction from './CiteModalAction';
+import ListItemAction from '../../common/components/ListItemAction';
+import Latex from '../../common/components/Latex';
+
+class LiteratureItem extends Component {
+  render() {
+    const { metadata, display } = this.props;
```
Comment (drjova) [inline src/literature/components/LiteratureItem.jsx]: do we need ``display`` anywhere?

File: src/literature/components/LiteratureItem.jsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    const abstract = metadata.getIn(['abstracts', 0, 'value']);
+
+    return (
+      <ResultItem
+        title={(
+          <Link to={`/literatue/${recordId}`}>
+            <Latex>{title}</Latex>
+          </Link>
+        )}
+        description={(
```
Comment (drjova) [inline src/literature/components/LiteratureItem.jsx]:
```js
description={[
   <AuthorList recordId={recordId} authors={authors} />,
   <LiteratureDate date={date} />
]}
```

File: src/common/components/Latex/__tests__/Latex.test.jsx (hunk 1 above)
Comment (drjova) [reply src/common/components/Latex/__tests__/Latex.test.jsx]: yep but just to make sure the style is remaining as expected with bigger text

**Pull request merged**